src/server/license/server/api_server.go

```diff
@@ -30,16 +33,79 @@ type apiServer struct {
 }
 
 // New returns an implementation of license.APIServer.
-func New(env Env) (lc.APIServer, error) {
+func New(env Env, public bool) (lc.APIServer, error) {
 	s := &apiServer{
 		env:     env,
 		license: licenseCollection(env.DB, env.Listener),
 	}
+	// envBootstrap runs concurrently because it reaches out to the enterprise service which heartbeats back to the license service. Therefore it must return to allow its service endpoints to be accessible
+	if !public {
+		go s.envBootstrap(context.Background())
+	}
 	return s, nil
 }
 
+func (a *apiServer) envBootstrap(ctx context.Context) error {
+	if a.env.Config.LicenseKey == "" && a.env.Config.EnterpriseSecret == "" {
+		return nil
+	}
+	if a.env.Config.LicenseKey == "" || a.env.Config.EnterpriseSecret == "" {
+		errors.New("License server failed to bootstrap via environment; Either both or neither of LICENSE_KEY and ENTERPRISE_SECRET must be set.")
+	}
+	logrus.Info("Started to configure license server via environment")
+	localhostPeerAddr := "grpc://localhost:" + fmt.Sprint(a.env.Config.PeerPort)
+	if err := func() error {
+		ctx = auth.AsInternalUser(ctx, "license-server")
+		_, err := a.activate(ctx, &lc.ActivateRequest{ActivationCode: a.env.Config.LicenseKey})
+		if err != nil {
+			return err
+		}
+		_, err = a.AddCluster(ctx, &lc.AddClusterRequest{
+			Id:               localhostEnterpriseClusterId,
+			Address:          localhostPeerAddr,
+			UserAddress:      localhostPeerAddr,
+			Secret:           a.env.Config.EnterpriseSecret,
+			EnterpriseServer: true,
+		})
+		if err != nil {
+			if errors.As(err, lc.ErrDuplicateClusterID) {
+				_, err = a.UpdateCluster(ctx, &lc.UpdateClusterRequest{
+					Id:          localhostEnterpriseClusterId,
+					Address:     localhostPeerAddr,
+					UserAddress: localhostPeerAddr,
+					Secret:      a.env.Config.EnterpriseSecret,
+				})
+				if err != nil {
+					return err
+				}
+			} else {
+				return err
+			}
+		}
+		_, err = a.env.EnterpriseServer.Activate(ctx, &ec.ActivateRequest{
+			Id:            localhostEnterpriseClusterId,
+			LicenseServer: localhostPeerAddr,
+			Secret:        a.env.Config.EnterpriseSecret})
+		if err != nil {
+			return errors.EnsureStack(err)
+		}
+		return nil
+	}(); err != nil {
+		panic(fmt.Sprintf("failed to configure the License Server from the environment: %v", err))
```

Comment:
Do we still want this to panic?  This can probably be de-indented and the individual things above can return errors, and where you write `go envBoostrap()` can instead be `go func() { err := envBootstrap(); if err != nil { panic(err) }`.